---
title: "_longevityTools_: Connecting Drug- and Age-related Gene Expression Signatures" 
author: "Authors: Tyler Backman, Dan Evans, Thomas Girke"
date: "Last update: `r format(Sys.time(), '%d %B, %Y')`" 
output:
  BiocStyle::html_document:
    toc: true
    toc_depth: 3
    fig_caption: yes

fontsize: 14pt
bibliography: bibtex.bib
---
<!--
%% \VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Overview Vignette}
%% \VignetteDepends{methods}
%% \VignetteKeywords{compute cluster, pipeline, reports}
%% \VignettePackage{longevityTools}
-->

<!---
- Compile from command-line
echo "rmarkdown::render('longevityTools_CMAP.Rmd')" | R -slave; R CMD Stangle longevityTools_CMAP.Rmd

- Commit to github
git commit -am "some edits"; git push -u origin master

- To customize font size and other style features, add this line to output section in preamble:  
    css: style.css
-->

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
options(width=100, max.print=1000)
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE")))
```

```{r setup, echo=FALSE, messages=FALSE, warnings=FALSE}
suppressPackageStartupMessages({
    library(longevityTools) 
    library(ggplot2) }) 
```

Note: the GitHub repository of this package is <a href="https://github.com/tgirke/longevityTools">here</a> and the most recent version of this tutorial can be found <a href="https://htmlpreview.github.io/?https://github.com/tgirke/longevityTools/blob/master/vignettes/longevityTools_CMAP.html">here</a>.

# Introduction 
This study analyzes drug- and age-related genome-wide expression data from public microarray 
and RNA-Seq experiments. One of the main objective is the identification drug candidates 
modulating the expression of longevity genes and pathways. For this, we compare age-related expression signatures with those from 
drug treamtments. The age-related query signatures are from recent publications such as 
Peters et al. [-@Peters2015-fc] and Sood et al. [-@Sood2015-pb], while the drug-related reference 
signatures are from the Connectivity Map (CMAP) and LINCS projects [@Lamb2006-uv].

<div align="right">[Back to Table of Contents]()</div>


# Getting Started

## Installation

The R software for running [_`longevityTools`_](https://github.com/tgirke/longevityTools) can be downloaded from [_CRAN_](http://cran.at.r-project.org/). The _`longevityTools`_ package can be installed from the R console using the following _`biocLite`_ install command. 

```{r install, eval=FALSE}
source("http://bioconductor.org/biocLite.R") # Sources the biocLite.R installation script 
biocLite("tgirke/longevityTools", build_vignettes=TRUE, dependencies=TRUE) # Installs package from GitHub
```
<div align="right">[Back to Table of Contents]()</div>

## Loading package and documentation

```{r documentation, eval=FALSE}
library("longevityTools") # Loads the package
library(help="longevityTools") # Lists package info
vignette(topic="longevityTools_CMAP", package="longevityTools") # Opens vignette
```
<div align="right">[Back to Table of Contents]()</div>

## Import custom functions 
Preliminary functions that are still under developement, or not fully tested and documented 
can be imported with the `source` command from the `inst/extdata` directory of the package.

```{r source_fct, eval=TRUE}
fctpath <- system.file("extdata", "analysis_Fct.R", package="longevityTools")
source(fctpath)
```

## Download data from Connectivity Map project site
The drug-related expression data were downloaded from the CMAP web site
[here](http://www.broadinstitute.org/cmap). The `getCmap` function downloads
the CMAP rank matrix and the compound annotation files, while `getCmapCEL`
downloads the corresponding 7,056 CEL files. The functions will write the
downloaded files to a `data` directory within the present workding directory of
the user's R session. This `data` directory plus a `data/CEL` sub-directory
needs to be created by the user prior to running the download functions.

```{r download_cmap, eval=TRUE}
getCmap(rerun=FALSE) # Downloads cmap rank matrix and compound annotation files
getCmapCEL(rerun=FALSE) # Download cmap CEL files. Note, this will take some time
```

<div align="right">[Back to Table of Contents]()</div>

# Funding
This project is funded by NIH grant U24AG051129 awarded by the National Intitute of Aging (NIA).

<div align="right">[Back to Table of Contents]()</div>

# Version information

```{r sessionInfo}
sessionInfo()
```
<div align="right">[Back to Table of Contents]()</div>

# References
