<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Authors: Thomas Girke, Danjuma Quarless, Tyler Backman, Kuan-Fu Ding, Jamison McCorrison, Nik Schork, Dan Evans" />


<title>longevityToolseDRUG - Connecting Drug- and Age-related Gene Expression Signatures</title>


<style type="text/css">code{white-space: pre;}</style>
<link href="data:text/css;charset=utf-8,pre%20%2Eoperator%2C%0Apre%20%2Eparen%20%7B%0Acolor%3A%20rgb%28104%2C%20118%2C%20135%29%0A%7D%0Apre%20%2Eliteral%20%7B%0Acolor%3A%20%23990073%0A%7D%0Apre%20%2Enumber%20%7B%0Acolor%3A%20%23099%3B%0A%7D%0Apre%20%2Ecomment%20%7B%0Acolor%3A%20%23998%3B%0Afont%2Dstyle%3A%20italic%0A%7D%0Apre%20%2Ekeyword%20%7B%0Acolor%3A%20%23900%3B%0Afont%2Dweight%3A%20bold%0A%7D%0Apre%20%2Eidentifier%20%7B%0Acolor%3A%20rgb%280%2C%200%2C%200%29%3B%0A%7D%0Apre%20%2Estring%20%7B%0Acolor%3A%20%23d14%3B%0A%7D%0A" rel="stylesheet" type="text/css" />
<script src="data:application/x-javascript;base64,
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();

"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<link href="data:text/css;charset=utf-8,body%2C%20td%20%7B%0Afont%2Dfamily%3A%20sans%2Dserif%3B%0Abackground%2Dcolor%3A%20white%3B%0Afont%2Dsize%3A%2013px%3B%0A%7D%0Abody%20%7B%0Amax%2Dwidth%3A%20800px%3B%0Amargin%3A%200%20auto%3B%0Apadding%3A%201em%201em%202em%3B%0Aline%2Dheight%3A%2020px%3B%0A%7D%0A%0Adiv%23TOC%20li%20%7B%0Alist%2Dstyle%3Anone%3B%0Abackground%2Dimage%3Anone%3B%0Abackground%2Drepeat%3Anone%3B%0Abackground%2Dposition%3A0%3B%0A%7D%0A%0Ap%2C%20pre%20%7B%20margin%3A%200em%200em%201em%3B%0A%7D%0A%0Aimg%2C%20table%20%7B%0Amargin%3A%200em%20auto%201em%3B%0A%7D%0Ap%20%7B%0Atext%2Dalign%3A%20justify%3B%0A%7D%0Att%2C%20code%2C%20pre%20%7B%0Afont%2Dfamily%3A%20%27DejaVu%20Sans%20Mono%27%2C%20%27Droid%20Sans%20Mono%27%2C%20%27Lucida%20Console%27%2C%20Consolas%2C%20Monaco%2C%20monospace%3B%0A%7D%0Ah1%2C%20h2%2C%20h3%2C%20h4%2C%20h5%2C%20h6%20%7B%20font%2Dfamily%3A%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Amargin%3A%201%2E2em%200em%200%2E6em%200em%3B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Ah1%2Etitle%20%7B%0Afont%2Dsize%3A%20250%25%3B%0Afont%2Dweight%3A%20normal%3B%0Acolor%3A%20%2387b13f%3B%0Aline%2Dheight%3A%201%2E1em%3B%0Amargin%2Dtop%3A%200px%3B%0Aborder%2Dbottom%3A%200px%3B%0A%7D%0Ah1%20%7B%0Afont%2Dsize%3A%20160%25%3B%0Afont%2Dweight%3A%20normal%3B%0Aline%2Dheight%3A%201%2E4em%3B%0Aborder%2Dbottom%3A%201px%20%231a81c2%20solid%3B%0A%7D%0Ah2%20%7B%0Afont%2Dsize%3A%20130%25%3B%20%7D%0Ah1%2C%20h2%2C%20h3%20%7B%0Acolor%3A%20%231a81c2%3B%0A%7D%0Ah3%2C%20h4%2C%20h5%2C%20h6%20%7B%0Afont%2Dsize%3A115%25%3B%0A%7D%20%0A%0Aa%20%7B%20color%3A%20%231a81c2%3B%20%7D%0Aa%3Aactive%20%7B%20outline%3A%20none%3B%20%7D%0Aa%3Avisited%20%7B%20color%3A%20%231a81c2%3B%20%7D%0Aa%3Ahover%20%7B%20color%3A%20%234c94c2%3B%20%7D%0Apre%2C%20img%20%7B%0Amax%2Dwidth%3A%20100%25%3B%0Adisplay%3A%20block%3B%0A%7D%0Apre%20%7B%0Aborder%3A%200px%20none%3B%0Abackground%2Dcolor%3A%20%23F8F8F8%3B%0Awhite%2Dspace%3A%20pre%3B%0Aoverflow%2Dx%3A%20auto%3B%0A%7D%0Apre%20code%20%7B%0Aborder%3A%201px%20%23aaa%20dashed%3B%0Abackground%2Dcolor%3A%20white%3B%0Adisplay%3A%20block%3B%0Apadding%3A%201em%3B%20color%3A%20%23111%3B%0Aoverflow%2Dx%3A%20inherit%3B%0A%7D%0A%0Apre%20code%5Bclass%5D%20%7B%0Abackground%2Dcolor%3A%20inherit%3B%0A%7D%0A%0Apre%5Bclass%5D%20code%20%7B%0Abackground%2Dcolor%3A%20inherit%3B%0A%7D%0A%0Acode%20%7B%20background%2Dcolor%3A%20transparent%3B%0Acolor%3A%20%2387b13f%3B%0Afont%2Dsize%3A%2092%25%3B%0A%7D%0A%0Atable%2C%20td%2C%20th%20%7B%0Aborder%3A%20none%3B%0Apadding%3A%200%200%2E5em%3B%0A%7D%0A%0Atbody%20tr%3Anth%2Dchild%28odd%29%20td%20%7B%0Abackground%2Dcolor%3A%20%23F8F8F8%3B%0A%7D%0Ablockquote%20%7B%0Acolor%3A%23666666%3B%0Amargin%3A0%3B%0Apadding%2Dleft%3A%201em%3B%0Aborder%2Dleft%3A%200%2E5em%20%23EEE%20solid%3B%0A%7D%0Ahr%20%7B%0Aheight%3A%200px%3B%0Aborder%2Dbottom%3A%20none%3B%0Aborder%2Dtop%2Dwidth%3A%20thin%3B%0Aborder%2Dtop%2Dstyle%3A%20dotted%3B%0Aborder%2Dtop%2Dcolor%3A%20%23999999%3B%0A%7D%0Aspan%2Eheader%2Dsection%2Dnumber%20%7B%0Apadding%2Dright%3A%201em%3B%0A%7D%0Aspan%2Etoc%2Dsection%2Dnumber%3A%3Aafter%20%7B%0Acontent%3A%20%22%20%22%3B%0Awhite%2Dspace%3A%20pre%3B%0A%7D%0A%40media%20print%20%7B%0A%2A%20%7B%0Abackground%3A%20transparent%20%21important%3B%0Acolor%3A%20black%20%21important%3B%0Afilter%3Anone%20%21important%3B%0A%2Dms%2Dfilter%3A%20none%20%21important%3B%0A%7D%0Abody%20%7B%0Afont%2Dsize%3A12pt%3B%0Amax%2Dwidth%3A100%25%3B%0A%7D%0Aa%2C%20a%3Avisited%20%7B%0Atext%2Ddecoration%3A%20underline%3B%0A%7D%0Ahr%20%7B%0Avisibility%3A%20hidden%3B%0Apage%2Dbreak%2Dbefore%3A%20always%3B%0A%7D%0Apre%2C%20blockquote%20%7B%0Apadding%2Dright%3A%201em%3B%0Apage%2Dbreak%2Dinside%3A%20avoid%3B%0A%7D%0Atr%2C%20img%20%7B%0Apage%2Dbreak%2Dinside%3A%20avoid%3B%0A%7D%0Aimg%20%7B%0Amax%2Dwidth%3A%20100%25%20%21important%3B%0A%7D%0A%40page%20%3Aleft%20%7B%0Amargin%3A%2015mm%2020mm%2015mm%2010mm%3B%0A%7D%0A%40page%20%3Aright%20%7B%0Amargin%3A%2015mm%2010mm%2015mm%2020mm%3B%0A%7D%0Ap%2C%20h2%2C%20h3%20%7B%0Aorphans%3A%203%3B%20widows%3A%203%3B%0A%7D%0Ah2%2C%20h3%20%7B%0Apage%2Dbreak%2Dafter%3A%20avoid%3B%0A%7D%0A%7D%0A" rel="stylesheet" type="text/css" />

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if(links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>

</head>

<body>


<div id="header">
<h1 class="title">longevityToolseDRUG - Connecting Drug- and Age-related Gene Expression Signatures</h1>
<h4 class="author"><em>Authors: Thomas Girke, Danjuma Quarless, Tyler Backman, Kuan-Fu Ding, Jamison McCorrison, Nik Schork, Dan Evans</em></h4>
<h4 class="date"><em>Last update: 06 March, 2016</em></h4>
</div>
<h4 class="package">Package version: <span style="font-weight: normal">longevityTools 1.0.3</span></h4>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#introduction"><span class="toc-section-number">1</span> Introduction</a></li>
<li><a href="#getting-started"><span class="toc-section-number">2</span> Getting Started</a><ul>
<li><a href="#installation"><span class="toc-section-number">2.1</span> Installation</a></li>
<li><a href="#loading-package-and-documentation"><span class="toc-section-number">2.2</span> Loading package and documentation</a></li>
</ul></li>
<li><a href="#setup-of-working-environment"><span class="toc-section-number">3</span> Setup of working environment</a><ul>
<li><a href="#import-custom-functions"><span class="toc-section-number">3.1</span> Import custom functions</a></li>
<li><a href="#create-expected-directory-structure"><span class="toc-section-number">3.2</span> Create expected directory structure</a></li>
</ul></li>
<li><a href="#data-downloads"><span class="toc-section-number">4</span> Data downloads</a><ul>
<li><a href="#download-data-from-connectivity-map-project-site"><span class="toc-section-number">4.1</span> Download data from Connectivity Map project site</a></li>
<li><a href="#overiew-of-cmap-data"><span class="toc-section-number">4.2</span> Overiew of CMAP data</a></li>
</ul></li>
<li><a href="#pre-processing-of-cel-files"><span class="toc-section-number">5</span> Pre-processing of CEL files</a><ul>
<li><a href="#determine-chip-type-from-cel-files"><span class="toc-section-number">5.1</span> Determine chip type from CEL files</a></li>
<li><a href="#normalization-of-cel-files"><span class="toc-section-number">5.2</span> Normalization of CEL files</a></li>
<li><a href="#combine-results-from-same-chip-type-in-single-data-frame"><span class="toc-section-number">5.3</span> Combine results from same chip type in single data frame</a></li>
<li><a href="#clean-up-of-intermediate-files"><span class="toc-section-number">5.4</span> Clean-up of intermediate files</a></li>
</ul></li>
<li><a href="#deg-analysis-with-limma"><span class="toc-section-number">6</span> DEG analysis with <code>limma</code></a><ul>
<li><a href="#generate-list-of-cel-names-defining-treatment-vs.control-comparisons"><span class="toc-section-number">6.1</span> Generate list of CEL names defining treatment vs. control comparisons</a></li>
<li><a href="#load-normalized-expression-data"><span class="toc-section-number">6.2</span> Load normalized expression data</a></li>
<li><a href="#deg-analysis-with-limma-1"><span class="toc-section-number">6.3</span> DEG analysis with <code>limma</code></a></li>
<li><a href="#obtain-annotation-information"><span class="toc-section-number">6.4</span> Obtain annotation information</a></li>
<li><a href="#transform-probe-set-to-gene-level-data"><span class="toc-section-number">6.5</span> Transform probe set to gene level data</a></li>
<li><a href="#number-of-degs-per-drug-treatment"><span class="toc-section-number">6.6</span> Number of DEGs per drug treatment</a></li>
<li><a href="#identify-deg-overlaps-with-peters-et-al.--peters2015-fc"><span class="toc-section-number">6.7</span> Identify DEG overlaps with Peters et al. <span class="citation">(2015)</span></a></li>
<li><a href="#identify-deg-overlaps-with-sood-et-al.--sood2015-pb"><span class="toc-section-number">6.8</span> Identify DEG overlaps with Sood et al. <span class="citation">(2015)</span></a></li>
<li><a href="#drugs-affecting-known-longevity-genes"><span class="toc-section-number">6.9</span> Drugs affecting known longevity genes</a></li>
</ul></li>
<li><a href="#connectivity-maps-enrichment-analysis"><span class="toc-section-number">7</span> Connectivity maps enrichment analysis</a><ul>
<li><a href="#query-drug-signatures"><span class="toc-section-number">7.1</span> Query drug signatures</a></li>
<li><a href="#query-disease-signatures"><span class="toc-section-number">7.2</span> Query disease signatures</a></li>
</ul></li>
<li><a href="#age-drug-network-analysis"><span class="toc-section-number">8</span> Age-drug network analysis</a></li>
<li><a href="#age-disease-network-analysis"><span class="toc-section-number">9</span> Age-disease network analysis</a></li>
<li><a href="#funding"><span class="toc-section-number">10</span> Funding</a></li>
<li><a href="#version-information"><span class="toc-section-number">11</span> Version information</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<!--
%% \VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Overview Vignette}
%% \VignetteDepends{methods}
%% \VignetteKeywords{compute cluster, pipeline, reports}
%% \VignettePackage{longevityTools}
-->
<!---
- Compile from command-line
echo "rmarkdown::render('longevityTools_eDRUG.Rmd', clean=F)" | R -slave; R CMD Stangle longevityTools_eDRUG.Rmd; Rscript ../md2jekyll.R longevityTools_eDRUG.knit.md 5

- Commit to github
git commit -am "some edits"; git push -u origin master

- To customize font size and other style features, add this line to output section in preamble:  
    css: style.css
-->
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This vignette is part of the NIA funded Longevity Genomics project. For more information on this project please visit its website <a href="http://www.longevitygenomics.org/projects/">here</a>. The GitHub repository of the corresponding R package is available <a href="https://github.com/tgirke/longevityTools">here</a> and the most recent version of this vignette can be found <a href="https://htmlpreview.github.io/?https://github.com/tgirke/longevityTools/blob/master/vignettes/longevityTools_eDRUG.html">here</a>.</p>
<p>The project component covered by this vignette analyzes drug- and age-related genome-wide expression data from public microarray and RNA-Seq experiments. One of the main objective is the identification drug candidates modulating the expression of longevity genes and pathways. For this, we compare age-related expression signatures with those from drug treamtments. The age-related query signatures are from recent publications such as Peters et al. <span class="citation">(2015)</span> and Sood et al. <span class="citation">(2015)</span>, while the drug-related reference signatures are from the Connectivity Map (CMAP) and LINCS projects <span class="citation">(Lamb et al. 2006)</span>.</p>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="getting-started" class="section level1">
<h1><span class="header-section-number">2</span> Getting Started</h1>
<div id="installation" class="section level2">
<h2><span class="header-section-number">2.1</span> Installation</h2>
<p>The R software for running <a href="https://github.com/tgirke/longevityTools"><em><code>longevityTools</code></em></a> can be downloaded from <a href="http://cran.at.r-project.org/"><em>CRAN</em></a>. The <em><code>longevityTools</code></em> package can be installed from the R console using the following <em><code>biocLite</code></em> install command.</p>
<pre class="r"><code>source(&quot;http://bioconductor.org/biocLite.R&quot;) # Sources the biocLite.R installation script 
biocLite(&quot;tgirke/longevityTools&quot;, build_vignettes=FALSE) # Installs package from GitHub</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="loading-package-and-documentation" class="section level2">
<h2><span class="header-section-number">2.2</span> Loading package and documentation</h2>
<pre class="r"><code>library(&quot;longevityTools&quot;) # Loads the package
library(help=&quot;longevityTools&quot;) # Lists package info
vignette(topic=&quot;longevityTools_eDRUG&quot;, package=&quot;longevityTools&quot;) # Opens vignette</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
</div>
<div id="setup-of-working-environment" class="section level1">
<h1><span class="header-section-number">3</span> Setup of working environment</h1>
<div id="import-custom-functions" class="section level2">
<h2><span class="header-section-number">3.1</span> Import custom functions</h2>
<p>Preliminary functions that are still under developement, or not fully tested and documented can be imported with the <code>source</code> command from the <code>inst/extdata</code> directory of the package.</p>
<pre class="r"><code>fctpath &lt;- system.file(&quot;extdata&quot;, &quot;longevityTools_eDRUG_Fct.R&quot;, package=&quot;longevityTools&quot;)
source(fctpath)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="create-expected-directory-structure" class="section level2">
<h2><span class="header-section-number">3.2</span> Create expected directory structure</h2>
<p>The following creates the directory structure expected by this workflow. Input data will be stored in the <code>data</code> directory and results will be written to the <code>results</code> directory. All paths are given relative to the present working directory of the user’s R session.</p>
<pre class="r"><code>dir.create(&quot;data&quot;); dir.create(&quot;data/CEL&quot;); dir.create(&quot;results&quot;) </code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
</div>
<div id="data-downloads" class="section level1">
<h1><span class="header-section-number">4</span> Data downloads</h1>
<div id="download-data-from-connectivity-map-project-site" class="section level2">
<h2><span class="header-section-number">4.1</span> Download data from Connectivity Map project site</h2>
<p>The drug-related expression data are downloaded from the CMAP web site <a href="http://www.broadinstitute.org/cmap">here</a>. The <code>getCmap</code> function downloads the CMAP rank matrix along with the compound annotations, and <code>getCmapCEL</code> downloads the corresponding 7,056 CEL files. The functions will write the downloaded files to the <code>data</code> and <code>data/CEL</code> directories within the present working directory of the user’s R session. Since some of the raw data sets are large, the functions will only rerun the download if the argument <code>rerun</code> is assigned <code>TRUE</code>. If the raw data are not needed then users can skip this time consuming download step and work with the preprocessed data obtained in the next section.</p>
<pre class="r"><code>getCmap(rerun=FALSE) # Downloads cmap rank matrix and compound annotation files
getCmapCEL(rerun=FALSE) # Download cmap CEL files. Note, this will take some time</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="overiew-of-cmap-data" class="section level2">
<h2><span class="header-section-number">4.2</span> Overiew of CMAP data</h2>
<p>The experimental design of the CMAP project is defined in the file <code>cmap_instances_02.xls</code>. Note, this file required some cleaning in LibreOffice (Excel would work for this too). After this it was saved as tab delimited txt file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/data/cmap_instances_02.txt">cmap_instances_02.txt</a>. The following count statisitics are extracted from this file.</p>
<p>The panel of cell lines used by CMAP includes <a href="http://www.broadinstitute.org/cmap/help_topics_linkified.jsp#mcf7">MCF7</a>, <a href="http://www.broadinstitute.org/cmap/help_topics_linkified.jsp#mcf7">ssMCF7</a>, <a href="http://www.broadinstitute.org/cmap/help_topics_linkified.jsp#hl60">HL60</a>, <a href="http://www.broadinstitute.org/cmap/help_topics_linkified.jsp#pc3">PC3</a> and <a href="http://www.broadinstitute.org/cmap/help_topics_linkified.jsp#skmel5">SKMEL5</a>. Each cell type was subjected to the following number of total treatments and number of distinct drugs, respectively. The total number of drugs used by CMAP is 1,309.</p>
<pre class="r"><code>library(ggplot2); library(reshape2) 
cmap &lt;- read.delim(&quot;./data/cmap_instances_02.txt&quot;, check.names=FALSE) 
df &lt;- data.frame(table(cmap[, &quot;cell2&quot;]), as.numeric(table(cmap[!duplicated(paste0(cmap$cmap_name, cmap$cell2)),&quot;cell2&quot;])))
colnames(df) &lt;- c(&quot;Cell_type&quot;, &quot;Treatments&quot;, &quot;Drugs&quot;)
df &lt;- melt(df, id.vars=c(&quot;Cell_type&quot;), variable.name = &quot;Samples&quot;, value.name=&quot;Counts&quot;)
ggplot(df, aes(Cell_type, Counts, fill=Samples)) + 
       geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;) + 
       ggtitle(&quot;Number of treatments by cell types&quot;)</code></pre>
<p>The number Affymetrix chip used in the experiments is plotted here for each of the three chip types used by CMAP:</p>
<pre class="r"><code>df &lt;- data.frame(table(cmap$array3)); colnames(df) &lt;- c(&quot;Chip_type&quot;, &quot;Counts&quot;) 
ggplot(df, aes(Chip_type, Counts)) + 
       geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;, fill=&quot;cornflowerblue&quot;) + 
       ggtitle(&quot;Number of chip types&quot;)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
</div>
<div id="pre-processing-of-cel-files" class="section level1">
<h1><span class="header-section-number">5</span> Pre-processing of CEL files</h1>
<div id="determine-chip-type-from-cel-files" class="section level2">
<h2><span class="header-section-number">5.1</span> Determine chip type from CEL files</h2>
<p>The CMAP data set is based on three different Affymetrix chip types (HG-U133A, HT_HG-U133A and U133AAofAv2). The following extracts the chip type information from the CEL files and stores the result in an <code>rds</code> file with the path <code>./data/chiptype.rds</code>. Users who skipped the download of the CEL files can download this file <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/data/chiptype.rds">here</a>.</p>
<pre class="r"><code>celfiles &lt;- list.files(&quot;./data/CEL&quot;, pattern=&quot;.CEL$&quot;)
chiptype &lt;- sapply(celfiles, function(x) affxparser::readCelHeader(paste0(&quot;data/CEL/&quot;, x))$chiptype)
if(FALSE) saveRDS(chiptype, &quot;./data/chiptype.rds&quot;) # if(FALSE) protects this line from accidental execution!</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="normalization-of-cel-files" class="section level2">
<h2><span class="header-section-number">5.2</span> Normalization of CEL files</h2>
<p>The follwoing processes the CEL files from each chip type separately using the MAS5 normalization algorithm. The results will be written to 3 subdirectores under <code>data</code> that are named after the chip type names. To save time, the processing is parallelized with <code>BiocParallel</code> to run on 100 CPU cores of a computer cluster with a scheduler (<em>e.g.</em> Torque). The number of CEL files from each chip type are: 807 CEL files from HG-U133A, 6029 CEL files from HT_HG-U133A, and 220 CEL files from U133AAofAv2. Note, these numbers are slightly different than those reported in the <code>cmap_instances_02.txt</code> file. The MAS5 normalized data sets can be downloaded here: <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/data/HG-U133A/all_mas5exprs.rds">HG-U133A</a>, <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/data/HT_HG-U133A/all_mas5exprs.rds">HT_HG-U133A</a>, <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/data/U133AAofAv2/all_mas5exprs.rds">U133AAofAv2</a>.</p>
<pre class="r"><code>library(BiocParallel); library(BatchJobs); library(affy)
chiptype &lt;- readRDS(&quot;./data/chiptype.rds&quot;)
chiptype_list &lt;- split(names(chiptype), as.character(chiptype))
normalizeCel(chiptype_list, rerun=FALSE) </code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="combine-results-from-same-chip-type-in-single-data-frame" class="section level2">
<h2><span class="header-section-number">5.3</span> Combine results from same chip type in single data frame</h2>
<pre class="r"><code>chiptype_dir &lt;- unique(readRDS(&quot;./data/chiptype.rds&quot;))
combineResults(chiptype_dir, rerun=FALSE)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="clean-up-of-intermediate-files" class="section level2">
<h2><span class="header-section-number">5.4</span> Clean-up of intermediate files</h2>
<p>This deletes intermediate files. Before executing these lines, please make sure that this is what you want.</p>
<pre class="r"><code>for(i in seq_along(chiptype_dir)) unlink(list.files(paste0(&quot;data/&quot;, chiptype_dir[i]), pattern=&quot;cellbatch&quot;, full.names=TRUE), recursive=TRUE)
unlink(&quot;data/CEL/*.CEL&quot;) # Deletes downloaded CEL files</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
</div>
<div id="deg-analysis-with-limma" class="section level1">
<h1><span class="header-section-number">6</span> DEG analysis with <code>limma</code></h1>
<div id="generate-list-of-cel-names-defining-treatment-vs.control-comparisons" class="section level2">
<h2><span class="header-section-number">6.1</span> Generate list of CEL names defining treatment vs. control comparisons</h2>
<p>The <code>sampleList</code> function extracts the sample comparisons (contrasts) from the CMAP annotation table and stores them as a list.</p>
<pre class="r"><code>cmap &lt;- read.delim(&quot;./data/cmap_instances_02.txt&quot;, check.names=FALSE) 
# comp_list &lt;- sampleList(cmap, myby=&quot;CMP&quot;)
comp_list &lt;- sampleList(cmap, myby=&quot;CMP_CELL&quot;)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="load-normalized-expression-data" class="section level2">
<h2><span class="header-section-number">6.2</span> Load normalized expression data</h2>
<p>The following loads the MAS5 normalized expression data into a single <code>data.frame</code>. To accelerate the import, the data is read from <code>rds</code> files.</p>
<pre class="r"><code>chiptype_dir &lt;- unique(readRDS(&quot;./data/chiptype.rds&quot;))
df1 &lt;- readRDS(paste0(&quot;data/&quot;, chiptype_dir[1], &quot;/&quot;, &quot;all_mas5exprs.rds&quot;))
df2 &lt;- readRDS(paste0(&quot;data/&quot;, chiptype_dir[2], &quot;/&quot;, &quot;all_mas5exprs.rds&quot;))
df3 &lt;- readRDS(paste0(&quot;data/&quot;, chiptype_dir[3], &quot;/&quot;, &quot;all_mas5exprs.rds&quot;))
affyid &lt;- rownames(df1)[rownames(df1) %in% rownames(df2)]; affyid &lt;- affyid[affyid %in% rownames(df3)]
mas5df &lt;- cbind(df1[affyid,], df2[affyid,], df3[affyid,])</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="deg-analysis-with-limma-1" class="section level2">
<h2><span class="header-section-number">6.3</span> DEG analysis with <code>limma</code></h2>
<p>The analysis of differentially expressed genes (DEGs) is performed with the <code>limma</code> package. Genes meeting the chosen cutoff criteria are reported as DEGs (below set to FDR of 10% and a minimum fold change of 2). The DEG matrix is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/degMA.xls"><code>degMA.xls</code></a>.</p>
<pre class="r"><code>degMA &lt;- runLimma(df=mas5df, comp_list, fdr=0.10, foldchange=1, verbose=TRUE, affyid=NULL)
degMA &lt;- degMA[ , !is.na(colSums(degMA))] # Remove columns where DEG analysis was not possible
write.table(degMA, file=&quot;./results/degMA.xls&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA) 
saveRDS(degMA, &quot;./results/degMA.rds&quot;)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="obtain-annotation-information" class="section level2">
<h2><span class="header-section-number">6.4</span> Obtain annotation information</h2>
<p>The following generates annotation information for the Affymetirx probe set identifiers. Note, the three different Affymetrix chip types used by CMAP share most probe set ids (&gt;95%), meaning it is possible to combine the data after normalization and use the same annotation package for all of them. The annotation libraries for the chip types HG-U133A and HT_HG-U133A are <code>hgu133a.db</code> and <code>hthgu133a.db</code>, respectively. However, there is no annotation library (e.g. CDF) available for U133AAofAv2.</p>
<pre class="r"><code>library(hgu133a.db)
myAnnot &lt;- data.frame(ACCNUM=sapply(contents(hgu133aACCNUM), paste, collapse=&quot;, &quot;), 
                             SYMBOL=sapply(contents(hgu133aSYMBOL), paste, collapse=&quot;, &quot;), 
                             UNIGENE=sapply(contents(hgu133aUNIGENE), paste, collapse=&quot;, &quot;), 
                             ENTREZID=sapply(contents(hgu133aENTREZID), paste, collapse=&quot;, &quot;), 
                             ENSEMBL=sapply(contents(hgu133aENSEMBL), paste, collapse=&quot;, &quot;), 
                             DESC=sapply(contents(hgu133aGENENAME), paste, collapse=&quot;, &quot;))
saveRDS(myAnnot, &quot;./results/myAnnot.rds&quot;)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="transform-probe-set-to-gene-level-data" class="section level2">
<h2><span class="header-section-number">6.5</span> Transform probe set to gene level data</h2>
<p>The <code>probeset2gene</code> function translates a binary DEG matrix from the probe level (<code>row.names</code> are Affy IDs) to the gene level (<code>row.names</code> are gene IDs). Genes represented by several probe sets (rows) will be collapsed based on a chosen summary rule. The following summary rules are supported: <code>summary_rule=1L</code> will consider a gene as a DEG for a specific treatment if at least one of several probe sets supports this assignment, while <code>summary_rule=2L</code> requires support from at least 2 probe sets.</p>
<pre class="r"><code>myAnnot &lt;- readRDS(&quot;./results/myAnnot.rds&quot;) 
degMA &lt;- readRDS(&quot;./results/degMA.rds&quot;) # Faster than read.delim()
degMAgene &lt;- probeset2gene(degMA, myAnnot, geneIDtype=&quot;ENTREZID&quot;, summary_rule=1L)
saveRDS(degMAgene, &quot;./results/degMAgene.rds&quot;)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="number-of-degs-per-drug-treatment" class="section level2">
<h2><span class="header-section-number">6.6</span> Number of DEGs per drug treatment</h2>
<pre class="r"><code>degMAgene &lt;- readRDS(&quot;./results/degMAgene.rds&quot;)
y &lt;- as.numeric(colSums(degMAgene))
interval &lt;- table(cut(y, right=FALSE, dig.lab=5,  breaks=c(0, 5, 10, 50, 100, 200, 500, 1000, 10000)))
df &lt;- data.frame(interval); colnames(df) &lt;- c(&quot;Bins&quot;, &quot;Counts&quot;)
ggplot(df, aes(Bins, Counts)) + 
       geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;, fill=&quot;cornflowerblue&quot;) + 
       ggtitle(&quot;DEG numbers by bins&quot;)</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="identify-deg-overlaps-with-peters-et-al.--peters2015-fc" class="section level2">
<h2><span class="header-section-number">6.7</span> Identify DEG overlaps with Peters et al. <span class="citation">(2015)</span></h2>
<p>Peters et al. <span class="citation">(2015)</span> reported 1,497 age-related gene expression signatures. The <code>intersectStats</code> function computes their intersects with each of the 3,318 drug-responsive DEG sets from CMAP. The result includes the Jaccard index as a simple similarity metric for gene sets as well as the raw and adjusted p-values based on the hypergeometric distribution expressing how likely it is to observe the observed intersect sizes just by chance. The results for the 20 top scoring drugs are given below and the full data set is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/degOL_PMID26490707.xls"><code>degOL_PMID26490707.xls</code></a>.</p>
<pre class="r"><code>PMID26490707 &lt;- read.delim(&quot;./data/PMID26490707_S1.xls&quot;, comment=&quot;#&quot;)
myAnnot &lt;- readRDS(&quot;./results/myAnnot.rds&quot;) 
geneid &lt;- as.character(PMID26490707$&quot;NEW.Entrez.ID&quot;)
degMAgene &lt;- readRDS(&quot;./results/degMAgene.rds&quot;) # Faster than read.delim()
degMAsub &lt;- degMAgene[rownames(degMAgene) %in% geneid,]
degOL_PMID26490707 &lt;- intersectStats(degMAgene, degMAsub)
write.table(degOL_PMID26490707, file=&quot;./results/degOL_PMID26490707.xls&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA) 
sum(degOL_PMID26490707[,1] &gt; 0) # Drugs with any overlap
degOL_PMID26490707[1:20,]</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="identify-deg-overlaps-with-sood-et-al.--sood2015-pb" class="section level2">
<h2><span class="header-section-number">6.8</span> Identify DEG overlaps with Sood et al. <span class="citation">(2015)</span></h2>
<p>Sood et al. <span class="citation">(2015)</span> reported 150 age-related gene expression signatures. The <code>intersectStats</code> function computes their intersects with each of the 3,318 drug-responsive DEG sets from CMAP. The result includes the Jaccard index as a simple similarity metric for gene sets as well as the raw and adjusted p-values based on the hypergeometric distribution expressing how likely it is to observe the observed intersect sizes just by chance. The results for the 20 top scoring drugs are given below and the full data set is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/degOL_PMID26343147.xls"><code>degOL_PMID26343147.xls</code></a>.</p>
<pre class="r"><code>PMID26343147 &lt;- read.delim(&quot;./data/PMID26343147_S1T1.xls&quot;, check.names=FALSE, comment=&quot;#&quot;)
myAnnot &lt;- readRDS(&quot;./results/myAnnot.rds&quot;) 
affyid &lt;- row.names(myAnnot[myAnnot[,&quot;SYMBOL&quot;] %in% PMID26343147[,&quot;Gene Symbol&quot;], ]) 
degMA &lt;- readRDS(&quot;./results/degMA.rds&quot;) # Faster then read.delim()
degMA &lt;- degMA[ , !is.na(colSums(degMA))] # Remove columns where DEG analysis was not possible
degMAsub &lt;- degMA[affyid,]
degOL_PMID26343147 &lt;- intersectStats(degMAgene, degMAsub)
write.table(degOL_PMID26343147, file=&quot;./results/degOL_PMID26343147.xls&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA) 
sum(degOL_PMID26343147[,1] &gt; 0) # Drugs with any overlap
degOL_PMID26343147[1:20,] # Top 20 scoring drugs</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="drugs-affecting-known-longevity-genes" class="section level2">
<h2><span class="header-section-number">6.9</span> Drugs affecting known longevity genes</h2>
<p>The following identifies CMAP drugs affecting the expression of the IGF1 or IGF1R genes. The final result is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/deg_IGF1.xls"><code>deg_IGF1.xls</code></a>.</p>
<pre class="r"><code>genesymbols &lt;- c(&quot;IGF1&quot;, &quot;IGF1R&quot;)
geneids &lt;- unique(as.character(myAnnot[myAnnot$SYMBOL %in% genesymbols,&quot;ENTREZID&quot;]))
names(geneids) &lt;- unique(as.character(myAnnot[myAnnot$SYMBOL %in% genesymbols,&quot;SYMBOL&quot;]))
degMAgene &lt;- readRDS(&quot;./results/degMAgene.rds&quot;) # Faster than read.delim()
df &lt;- data.frame(row.names=colnames(degMAgene), check.names=FALSE)
for(i in seq_along(geneids)) df &lt;- cbind(df, as.numeric(degMAgene[geneids[i],]))
colnames(df) &lt;- names(geneids)
df &lt;- df[rowSums(df)&gt;0,]
nrow(df) # Number of drugs affecting at least one of: IGF1 or IGF1R</code></pre>
<p>The following computes the <code>limma</code> FDR values for the corresponding genes (here IGF1 and IGF1R) and drug treatments.</p>
<pre class="r"><code>affyids2 &lt;- row.names(myAnnot[myAnnot$SYMBOL %in% genesymbols,])
affyids &lt;- as.character(myAnnot[myAnnot$SYMBOL %in% genesymbols,&quot;SYMBOL&quot;])
names(affyids) &lt;- affyids2
cmap &lt;- read.delim(&quot;./data/cmap_instances_02.txt&quot;, check.names=FALSE) 
comp_list &lt;- sampleList(cmap, myby=&quot;CMP_CELL&quot;)
comp_list &lt;- comp_list[row.names(df)]
degList &lt;- runLimma(df=mas5df, comp_list, fdr=0.10, foldchange=1, verbose=FALSE, affyid=names(affyids))
pvalDF &lt;- sapply(unique(affyids), function(x) sapply(rownames(df), function(y) min(degList[[y]][affyids==x,&quot;adj.P.Val&quot;])))
colnames(pvalDF) &lt;- paste0(colnames(pvalDF), &quot;_FDR&quot;)
df &lt;- cbind(df, pvalDF)
write.table(df, file=&quot;./results/deg_IGF1.xls&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA) </code></pre>
<p>Now the final <code>data.frame</code> can be sorted by increasing mean FDR values.</p>
<pre class="r"><code>igfDF &lt;- read.delim(&quot;./results/deg_IGF1.xls&quot;, row.names=1)
igfDF[order(rowMeans(igfDF[,3:4])),][1:20,]</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
</div>
<div id="connectivity-maps-enrichment-analysis" class="section level1">
<h1><span class="header-section-number">7</span> Connectivity maps enrichment analysis</h1>
<p>The connectivity maps approach is a rank-based enrichment method utilizing the KS test <span class="citation">(Lamb et al. 2006)</span>. It measures the similarities of expression signatures based on the enrichment of up- and down-regulated genes at the top and bottom of sorted (ranked) gene lists.</p>
<div id="query-drug-signatures" class="section level2">
<h2><span class="header-section-number">7.1</span> Query drug signatures</h2>
<p>The following uses the 1,497 age-related gene expression signatures from Peters et al. <span class="citation">(2015)</span> as a query against the CMAP signatures. The results are sorted by the ES Distance and the top scoring 20 drugs are given below. The full result table is<br />
written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/drugcmap2.xls"><code>drugcmap2.xls</code></a>.</p>
<pre class="r"><code>library(DrugVsDisease)
PMID26490707 &lt;- read.delim(&quot;./data/PMID26490707_S1.xls&quot;, comment=&quot;#&quot;, check.names=FALSE)
data(drugRL)
PMID26490707sub &lt;- PMID26490707[PMID26490707[,&quot;NEW-Gene-ID&quot;] %in% rownames(drugRL),]
testprofiles &lt;- list(ranklist=matrix(PMID26490707sub$Zscore, dimnames=list(PMID26490707sub[,&quot;NEW-Gene-ID&quot;])), 
                     pvalues=matrix(PMID26490707sub$P, dimnames=list(PMID26490707sub[,&quot;NEW-Gene-ID&quot;])))
drugcmap &lt;- classifyprofile(data=testprofiles$ranklist, case=&quot;disease&quot;, signif.fdr=0.5, no.signif=20)
drugcmap2 &lt;- classifyprofile(data=testprofiles$ranklist, case=&quot;disease&quot;, 
                            pvalues=testprofiles$pvalues, cytoout=FALSE, type=&quot;dynamic&quot;, 
                            dynamic.fdr=0.5, signif.fdr=0.05, adj=&quot;BH&quot;, no.signif=100)
write.table(drugcmap2, file=&quot;./results/drugcmap2.xls&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA) 
drugcmap2[[1]][1:20,]</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="query-disease-signatures" class="section level2">
<h2><span class="header-section-number">7.2</span> Query disease signatures</h2>
<p>The same query is performed against a reference set of disease expression signatures. The results are sorted by the ES Distance and the top scoring 20 drugs are given below. The full result table is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/diseasecmap2.xls"><code>diseasecmap2.xls</code></a>.</p>
<pre class="r"><code>PMID26490707 &lt;- read.delim(&quot;./data/PMID26490707_S1.xls&quot;, comment=&quot;#&quot;, check.names=FALSE)
data(diseaseRL)
PMID26490707sub &lt;- PMID26490707[PMID26490707[,&quot;NEW-Gene-ID&quot;] %in% rownames(diseaseRL),]
testprofiles &lt;- list(ranklist=matrix(PMID26490707sub$Zscore, dimnames=list(PMID26490707sub[,&quot;NEW-Gene-ID&quot;])), 
                     pvalues=matrix(PMID26490707sub$P, dimnames=list(PMID26490707sub[,&quot;NEW-Gene-ID&quot;])))
diseasecmap &lt;- classifyprofile(data=testprofiles$ranklist, case=&quot;drug&quot;, signif.fdr=0.5, no.signif=20)
diseasecmap2 &lt;- classifyprofile(data=testprofiles$ranklist, case=&quot;drug&quot;, 
                            pvalues=testprofiles$pvalues, cytoout=FALSE, type=&quot;dynamic&quot;, 
                            dynamic.fdr=0.5, adj=&quot;BH&quot;, no.signif=100)
write.table(diseasecmap2, file=&quot;./results/diseasecmap2.xls&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA) 
diseasecmap2[[1]][1:20,]</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
</div>
<div id="age-drug-network-analysis" class="section level1">
<h1><span class="header-section-number">8</span> Age-drug network analysis</h1>
<p>In progress…</p>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="age-disease-network-analysis" class="section level1">
<h1><span class="header-section-number">9</span> Age-disease network analysis</h1>
<p>In progress…</p>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="funding" class="section level1">
<h1><span class="header-section-number">10</span> Funding</h1>
<p>This project is funded by NIH grant U24AG051129 awarded by the National Intitute on Aging (NIA).</p>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="version-information" class="section level1">
<h1><span class="header-section-number">11</span> Version information</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.2.3 (2015-12-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.3 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  utils     datasets  grDevices methods   base     
## 
## other attached packages:
## [1] ggplot2_2.0.0        longevityTools_1.0.3 BiocStyle_1.8.0     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.3      digest_0.6.9     plyr_1.8.3       grid_3.2.3       gtable_0.1.2    
##  [6] formatR_1.2.1    magrittr_1.5     evaluate_0.8     scales_0.3.0     stringi_1.0-1   
## [11] rmarkdown_0.9.2  tools_3.2.3      stringr_1.0.0    munsell_0.4.2    yaml_2.1.13     
## [16] colorspace_1.2-6 htmltools_0.3    knitr_1.12</code></pre>
<div align="right">
<a href>Back to Table of Contents</a>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Lamb2006-uv">
<p>Lamb, J, E D Crawford, D Peck, J W Modell, I C Blat, M J Wrobel, J Lerner, et al. 2006. “The Connectivity Map: Using Gene-Expression Signatures to Connect Small Molecules, Genes, and Disease.” <em>Science</em> 313 (5795): 1929–35. doi:<a href="https://doi.org/10.1126/science.1132939">10.1126/science.1132939</a>.</p>
</div>
<div id="ref-Peters2015-fc">
<p>Peters, Marjolein J, Roby Joehanes, Luke C Pilling, Claudia Schurmann, Karen N Conneely, Joseph Powell, Eva Reinmaa, George L Sutphin,..., and Andrew D Johnson. 2015. “The Transcriptional Landscape of Age in Human Peripheral Blood.” <em>Nat. Commun.</em> 6 (22~oct): 8570. doi:<a href="https://doi.org/10.1038/ncomms9570">10.1038/ncomms9570</a>.</p>
</div>
<div id="ref-Sood2015-pb">
<p>Sood, Sanjana, Iain Gallagher, Katie Lunnon, Eric Rullman, Aoife Keohane, Hannah Crossland, Bethan Phillips, et al. 2015. “A Novel Multi-Tissue RNA Diagnostic of Healthy Ageing Relates to Cognitive Health Status.” <em>Genome Biology</em>. doi:<a href="https://doi.org/10.1186/s13059-015-0750-x">10.1186/s13059-015-0750-x</a>.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
